custom/icds_reports/utils.py

```diff
@@ -44,7 +44,7 @@ def __init__(self, domain, filters, report_id):
         self.report_config = report_config
 
     def data(self):
-        return self.report_config.get_data()
+        return self.report_config.data_source._get_data().values()
```

Comment:
@emord 
We can't add pagination to these reports because ```Block ASR``` has several reports with defined rows. I think, we have problem because we using  User Custom Report (not data sources) and we must aggregate data to correct format in python, thats why we must load all data.

I think to improve these reports we should refactor both 'Block' report and move aggregation to the SQL.